### ISSUE TITLE
java.lang.ClassCastException Camunda BPM 7.12.0 Keycloak 9.0.3 camunda-bpm-identity-keycloak 1.4.0

### ISSUE THREAD

**Issue by chaitanyabjoshi** [iceman91176/camunda-sso-tomcat#5]
Hello,

First of all thanks a lot for submitting this community PlugIn for Tomcat - Keycloak SSO. There are not many solutions about it.
However, while applying this PlugIn for Camunda 7.12.0 and Keycloak 9.0.3, I am getting following exception.

```
javax.servlet.ServletException: Cannot instantiate authentication filter: authentication provider does not implement interface org.camunda.bpm.engine.rest.security.auth.AuthenticationProvider
		at org.camunda.bpm.webapp.impl.security.auth.ContainerBasedAuthenticationFilter.init(ContainerBasedAuthenticationFilter.java:71)
...
Caused by: java.lang.ClassCastException: class org.camunda.community.auth.keycloak.sso.KeycloakSSOAuthenticationProvider cannot be cast to class org.camunda.bpm.engine.rest.security.auth.AuthenticationProvider (org.camunda.community.auth.keycloak.sso.KeycloakSSOAuthenticationProvider is in unnamed module of loader java.net.URLClassLoader @56235b8e; org.camunda.bpm.engine.rest.security.auth.AuthenticationProvider is in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @d963c85)
		at org.camunda.bpm.webapp.impl.security.auth.ContainerBasedAuthenticationFilter.init(ContainerBasedAuthenticationFilter.java:62)
```

Apart from updating dependencies versions, I have not made a single line change in the filter code itself. I have already got camunda-bpm-identity-keycloak Plugin working and I can read groups and users from keycloak without any problem.
I have attached the log for the exception here. [localhost.2020-04-22.log](https://github.com/iceman91176/camunda-bpm-auth-keycloak-sso/files/4517792/localhost.2020-04-22.log)

I am using dockerised environment if it's of any relevance. Thanks in advance for any help.

Best Regards
Chaitanya

**Comment by chaitanyabjoshi** [comment]
Update: I checked issue #2 but my web.xml configuration is exactly like the updated one and Tomcat seems to find the classes. So configuration is not really the problem.
File `keycloak.json` is in `$CATALINA_HOME/conf` directory.

**Comment by iceman91176** [comment]
Hello @chaitanyabjoshi ,
sorry for the very long delay. I don't to know if this of any relevance for you, but I did a rework of the library, which is not merged into master yet. I bumped it to Keycloak 11, Camunda 7.14 and camunda-bpm-identity-keycloak 2.0.0. 
I am not experiencing any issues - i used the same web.xml files for engine-rest and the webapps as before. There are almost no changes to the code-base itself.

In the upcoming release there will be a Docker-Build also, that puts together all the manual steps.
I have to adjust the README a little bit before merging it into master
